README.md

```diff
@@ -7,3 +7,30 @@ This plugin provides APIs that are used by core Pipeline plugins for features su
 ## Version History
 
 See [the changelog](CHANGELOG.md).
+
+## Caveat emptor
+
+### Overheads of WithThreadName feature
+
+Some consumers of this plugin improve readability of JVM thread dumps by using
+the `org.jenkinsci.plugins.workflow.support.concurrent.WithThreadName` feature
+to temporarily clarify the purpose of that thread.
+
+This ends up calling `Thread.setName()` => `JVM_SetNativeThreadName()` => some
+native OS implementation (via `pthreads` or otherwise), both to enter and exit
+a context with this feature.
+
+In some use-cases, e.g. with a generated `parallel` pipeline with hundreds of
+stages then waiting in a Jenkins queue for considerable time, this can amount
+to tens of thousands of native (not very efficient on some systems) calls per
+second just for this troubleshooting aid, and can cause considerable slow-down
+of the Jenkins controller.
+
+If you suspect your controller's performance is impacted by this, please
+configure a JVM property or context setting
```

Comment:
Fine though I would still rather just see this done automatically than offer yet another thing for users to think about.

Reply:
ESCOPECREEP :) That could be a next feature, if someone takes on it, and designs in a way that is easy to use and/or least-surprise for existing users.

If all this logging was introduced recently (I don't know) then the least-surprise could in fact be having that disabled by default and for the few developers/troubleshooters relying on nicer thread names to enable it somehow. At least now it is using not directly the JVM option but a Jenkins context reader that taps into that among other sources, thanks to you. Maybe there's a way to flip the switch while live?